Quadriga/src/main/java/edu/asu/spring/quadriga/service/resolver/impl/ProjectHandleResolverManager.java

```diff
@@ -43,32 +45,61 @@ public boolean saveProjectHandleResolver(IProjectHandleResolver resolver, String
             resolver.setId(id);
             resolver.setUsername(username);
         }
-        
+
         ProjectHandleResolverDTO resolverDto = mapper.mapProjectHandleResolver(resolver);
         resolverDao.saveOrUpdateDTO(resolverDto);
-        
+
         return true;
     }
-    
-    /* (non-Javadoc)
-     * @see edu.asu.spring.quadriga.service.resolver.impl.IProjectHandleResolverManager#getProjectHandleResolver(java.lang.String)
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.spring.quadriga.service.resolver.impl.
+     * IProjectHandleResolverManager#getProjectHandleResolver(java.lang.String)
      */
     @Override
     public IProjectHandleResolver getProjectHandleResolver(String id) {
         ProjectHandleResolverDTO dto = resolverDao.getDTO(id);
         return mapper.mapProjectHandleResolver(dto);
     }
-    
-    /* (non-Javadoc)
-     * @see edu.asu.spring.quadriga.service.resolver.impl.IProjectHandleResolverManager#getProjectHandleResolvers(java.lang.String)
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.spring.quadriga.service.resolver.impl.
+     * IProjectHandleResolverManager#getProjectHandleResolvers(java.lang.String)
      */
     @Override
     public List<IProjectHandleResolver> getProjectHandleResolvers(String username) {
         List<ProjectHandleResolverDTO> resolverDtos = resolverDao.getProjectResolversForUser(username);
-        
+
         List<IProjectHandleResolver> resolvers = new ArrayList<IProjectHandleResolver>();
         resolverDtos.forEach(dto -> resolvers.add(mapper.mapProjectHandleResolver(dto)));
-        
+
         return resolvers;
     }
+
+    /**
+     * Checks whether the given resolver is used by any project and if it's
+     * true, throws exception indicating the error. If no project is using the
+     * resolver, then the resolver is deleted from the database.
+     * 
+     * @param resolver
+     *            The project handle resolver to delete.
+     * @throws QuadrigaException
+     */
+    @Override
+    @Transactional
+    public boolean deleteProjectHandleResolver(IProjectHandleResolver resolver) throws QuadrigaException {
+
+        if (resolverDao.getProjectsForResolverId(resolver.getId()).size() > 0) {
+            throw new QuadrigaException("resolver cannot be deleted as it is being used by project");
```

Comment:
Also, don't just throw a quadrigaException. Use some specific (on the DAO) that can be handled appropriately in higher levels.